typing/typecore.ml

```diff
@@ -4643,11 +4656,11 @@ and type_application env funct sargs =
                 if not optional && is_optional l' then
                   Location.prerr_warning sarg.pexp_loc
                     (Warnings.Nonoptional_label (Printtyp.string_of_label l));
-                remaining_sargs, use_arg sarg l'
+                remaining_sargs, Some (use_arg sarg l', sarg.pexp_loc)
             | None ->
                 sargs,
                 if optional && List.mem_assoc Nolabel sargs then
-                  eliminate_optional_arg ()
+                  Some (eliminate_optional_arg (), Location.none)
```

Comment:
Using `Location.none` here is valid due to a non-local invariant: we are only using the location of the last argument and the test above  (`List.mem_assoc Nolabel sargs`) guarantees that there are more arguments after the current one. I would recommend to at least add a comment that we can use `Location.none` here because we know that there would be at least one argument with a valid location in `sargs`.

Reply:
I wasn't sure how to define that clearly so I switching to using an option. Not all argument have a location.